CONTRIBUTING.md

```diff
@@ -6,29 +6,300 @@ This page is intended for contributors to the registry and packages.
+    per dataset).
+
+8. Define all variable properties.
+
+    The variable properties customize visualization of configuration options in the Kibana UI. Make sure they're
+    defined in all manifest files.
+
+```yaml
+    vars:
+      - name: paths
+        required: true
+        show_user: true
+        title: Access log paths
+        description: Paths to the nginx access log file.
+        type: text
+        multi: true
+        default:
+          - /var/log/nginx/access.log*
+```
+
+**required** - option is required
+
+**show_user** - don't hide the configuration option (collapsed menu)
+
+**title** - human readable variable name
+
+**description** - variable description (may contain some details)
+
+**type** - field type (according to the reference: text, password, bool, integer)
+
+**multi** - the field has mutliple values.
+
+9. Update docs template with sample events.
+
+    The events collected by the agent slightly differ from original, Metricbeat's and Filebeat's, ones. Adjust the event
+    content manually basing on already migrated integrations (e.g. [MySQL integration](https://github.com/elastic/package-registry/tree/master/dev/import-beats-resources/mysql/docs))
+    or copy them once managed to run whole setup with real agent.
+
+## Testing and validation
+
+### Run the whole setup
+
+1. Build docker image with EPR and run it:
+
+    ```bash
+   $ docker build --rm -t integrations_registry:latest .
+   $ docker run -i -t -p 8080:8080 $(docker images -q integrations_registry:latest) 
+   ```
+
+3. Verify that your integration is available (in the right version), e.g. MySQL: http://localhost:8080/search?package=mysql
+
+    ```json
+    [
+      {
+        "description": "MySQL Integration",
+        "download": "/epr/mysql/mysql-0.0.1.tar.gz",
+        "icons": [
+          {
+            "src": "/package/mysql/0.0.1/img/logo_mysql.svg",
+            "title": "logo mysql",
+            "size": "32x32",
+            "type": "image/svg+xml"
+          }
+        ],
+        "name": "mysql",
+        "path": "/package/mysql/0.0.1",
+        "title": "MySQL",
+        "type": "integration",
+        "version": "0.0.1"
+      }
+    ]
+    ```
+
+4. Start Elasticsearch instance:
+
+    ```bash
+   $ yarn es snapshot -E xpack.security.authc.api_key.enabled=true 
+   ```
+
+5. Start Kibana with enabled Ingest Manager:
+
+    ```bash
+   $ yarn start --xpack.ingestManager.enabled=true --xpack.ingestManager.epm.enabled=true --xpack.ingestManager.fleet.enabled=true --xpack.ingestManager.epm.registryUrl=http://localhost:8080/
+    ```
+
+6. Build agent code:
+    ```bash
+   $ cd $GOPATH/src/github.com/elastic/beats/x-pack/elastic-agent
+   $ PLATFORMS=darwin mage package
+    ```
+
+   Unpack the distribution you'd like to use (e.g. tar.gz):
+   ```bash
+   $ cd build/distributions/
+   $ tar xzf elastic-agent-8.0.0-darwin-x86_64.tar.gz
+   $ cd elastic-agent-8.0.0-darwin-x86_64/
+   ```
```

Comment:
I think this is fine, wondering if we have a snapshot image being produced already? cc @ph @ruflin

Reply:
I will take a look and get back to you @exekias